packages/runtime/container-runtime/src/gc/gcSummaryStateTracker.ts

```diff
@@ -276,38 +290,38 @@ export class GCSummaryStateTracker {
 		// If the summary was not tracked by this client, the state should be updated from the downloaded snapshot.
 		const snapshotTree = result.snapshotTree;
 		const metadataBlobId = snapshotTree.blobs[metadataBlobName];
-		if (metadataBlobId) {
-			const metadata = await readAndParseBlob<IContainerRuntimeMetadata>(metadataBlobId);
-			this.latestSummaryGCVersion = getGCVersion(metadata);
-		}
+		const metadata = metadataBlobId
+			? await readAndParseBlob<IContainerRuntimeMetadata>(metadataBlobId)
+			: undefined;
+		this.latestSummaryGCVersion = getGCVersion(metadata);
 
 		const gcSnapshotTree = snapshotTree.trees[gcTreeKey];
 		// If GC ran in the container that generated this snapshot, it will have a GC tree.
 		this.wasGCRunInLatestSummary = gcSnapshotTree !== undefined;
-		let latestGCData: IGarbageCollectionSnapshotData | undefined;
-		if (gcSnapshotTree !== undefined) {
-			latestGCData = await getGCDataFromSnapshot(gcSnapshotTree, readAndParseBlob);
+
+		if (gcSnapshotTree === undefined) {
+			return undefined;
 		}
-		this.pendingSummaryData = undefined;
-		return latestGCData;
-	}
 
-	/**
-	 * Update state from the given snapshot data. This is done during load and during refreshing state from a snapshot.
-	 * @param gcSnapshotData - The GC snapshot data to update state from.
-	 */
-	public updateStateFromSnapshotData(gcSnapshotData: IGarbageCollectionSnapshotData | undefined) {
-		// If there is no snapshot data, it means this snapshot was generated with GC disabled. Unset all GC state.
```

Comment:
Is this case lost for places that used to call `updateStateFromSnapshotData`?

Reply:
This happens in `refreshLatestSummary` or in `initializeBaseState` making it more explicit rather than a generic update state method.